types/knockout/index.d.ts

```diff
@@ -946,15 +947,16 @@ interface KnockoutComputedContext {
 // and used Union Types to simplify overloads (requires TypeScript 1.4)
 //
 declare namespace KnockoutComponentTypes {
-
-    interface Config {
-        viewModel?: ViewModelFunction | ViewModelSharedInstance | ViewModelFactoryFunction | AMDModule;
+    type ViewModel = ViewModelFunction | ViewModelSharedInstance | ViewModelFactoryFunction | AMDModule;
+    
+    interface Config<T> {
```

Comment:
Should there be some restriction on `T` here?  It seems weird that you could have something like `viewModel: string`, but maybe that is reasonable?  (I haven't actually used KO's component APIs, myself)

Reply:
I couldn't find a better way to do this, and this will allow specifying the class allowing to be more strict.
this is the is idea whit this change:
`
components.register<ComponentClass>('component-name', {
    viewModel: ComponentClass,
    template: require('./containers/units-set-list/units-set-list-component.html')
});
`